File: StepByStep/Brute force/1018_체스판다시칠하기.py
row, col = map(int, input().split())
board = []
answer = row*col
# board판 입력 받기
for _ in range(row):
    board.append(input())
whiteFirst = ["WBWBWBWB",
              "BWBWBWBW",
              "WBWBWBWB",
              "BWBWBWBW",
              "WBWBWBWB",
              "BWBWBWBW",
              "WBWBWBWB",
              "BWBWBWBW"]
blackFirst = ["BWBWBWBW",
              "WBWBWBWB",
              "BWBWBWBW",
              "WBWBWBWB",
              "BWBWBWBW",
              "WBWBWBWB",
              "BWBWBWBW",
              "WBWBWBWB"]

for i in range(row-7):
    for j in range(col-7):
        cnt_b = 0
        cnt_w = 0
        for k in range(8):
            for m in range(8):
                # black이 먼저인 기준
                if board[i+k][j+m] != blackFirst[k][m]:
                    cnt_b += 1
                # white가 먼저인 기준
                if board[i+k][j+m] != whiteFirst[k][m]:
                    cnt_w += 1
        cnt = cnt_b if cnt_b < cnt_w else cnt_w
        if cnt < answer:
            answer = cnt
print(answer)

File: solved_ac/Class_2/1018_체스판다시칠하기.cpp
#include <algorithm>
#include <iostream>
#include <string>

using namespace std;

string white[8] = {"WBWBWBWB", "BWBWBWBW", "WBWBWBWB", "BWBWBWBW", "WBWBWBWB", "BWBWBWBW", "WBWBWBWB", "BWBWBWBW"};
string black[8] = {"BWBWBWBW", "WBWBWBWB", "BWBWBWBW", "WBWBWBWB", "BWBWBWBW", "WBWBWBWB", "BWBWBWBW", "WBWBWBWB"};
string input[50];
int n, m;
int ans = 2500;

int main()
{
    cin >> n >> m;
    for (int i = 0; i < n; i++)
        cin >> input[i];

    for (int i = 0; i < n - 7; i++)
    {
        for (int j = 0; j < m - 7; j++)
        {
            int cnt_black = 0;
            int cnt_white = 0;
            for (int r = 0; r < 8; r++)
            {
                for (int c = 0; c < 8; c++)
                {
                    if (white[r][c] != input[i + r][j + c])
                        cnt_white++;
                    if (black[r][c] != input[i + r][j + c])
                        cnt_black++;
                }
            }
            ans = min(ans, min(cnt_white, cnt_black));
        }
    }

    cout << ans << '\n';
    return 0;
}